### PR TITLE
ci: move from travis to github action

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v2
+        id: semantic
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          GIT_AUTHOR_EMAIL: ${{ secrets.GIT_AUTHOR_EMAIL }}
+          GIT_AUTHOR_NAME: ${{ secrets.GIT_AUTHOR_NAME }}
+          GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMITTER_EMAIL }}
+          GIT_COMMITTER_NAME: ${{ secrets.GIT_COMMITTER_NAME }}
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
```
Comment (arnaudbesnier) [inline .github/workflows/build.yml:87]: `NPM_TOKEN` is missing to publish the package.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Start docker container
+        run: docker-compose up -d
+      - name: Send coverage
+        uses: paambaati/codeclimate-action@v2.7.4
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
```
Comment (arnaudbesnier) [inline .github/workflows/build.yml:56]: Where is used the `CC_TEST_REPORTER_URL` to download the reporter we want (like it was done with TravisCI)?

**Review by arnaudbesnier — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (VincentMolinie) [reply .github/workflows/build.yml:56]: I don't understand what you mean. I looked into CC doc there is no `CC_TEST_REPORTER_URL`. I don't know for what it was used for 🤷  ?
Can you tell me a little bit more 🤔  ?
What do you mean by downloading the reporter we want ?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        with:
+          coverageCommand: yarn test:coverage
+
+  deploy:
+    name: Release package
+    runs-on: ubuntu-latest
+    needs: [test]
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/beta')
```
Comment (arnaudbesnier) [inline .github/workflows/build.yml:64]: Will it be played twice without the "[skip ci]" condition (on commit merge + release commit)?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v1
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Start docker container
```
Comment (arnaudbesnier) [inline .github/workflows/build.yml:51]: I think there is now a Docker Hub login in the master branch, not to reach the container downloads limits.
I guess you'll have to apply it here too.

**Review by arnaudbesnier — APPROVED** [review]

**Review by arnaudbesnier — APPROVED** [review]

**Pull request merged**